src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -0,0 +1,173 @@
+        }
+
+        return $portal->sqlData['main_node_tree'];
+    }
+
+    private function createTreeDataModel(\TdbCmsTree $node, string $fieldName, $path = ''): BackendTreeNodeDataModel
+    {
+        $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
+        $children = $node->GetChildren(true);
+
+        $path = $this->addBreadcrumbHtmlForNode($node, $fieldName, $path);
+
+        while ($child = $children->Next()) {
+            $childTreeNodeObj = $this->createTreeDataModel($child, $fieldName, $path);
+            $treeNodeDataModel->addChildren($childTreeNodeObj);
+        }
+
+        return $treeNodeDataModel;
+    }
+
+    protected function addBreadcrumbHtmlForNode(\TdbCmsTree $node, string $fieldName, string $path = ''): string
+    {
+        $path .= '<li class="breadcrumb-item">'.$node->fieldName."</li>\n";
+        $this->treePathHTML .= '<div id="'.$fieldName.'_tmp_path_'.$node->id.'" style="display:none;"><ol class="breadcrumb ml-0"><li class="breadcrumb-item"><i class="fas fa-sitemap"></i></li>'.$path.'</ol></div>'."\n";
+
+        return $path;
+    }
+
+    /**
+     * Fetches a list of all restricted nodes (of all portals).
+     * A restricted node is a startnavigation nodes of a portal.
+     */
+    private function getPortalNavigationStartNodes(): array
+    {
+        $portalList = \TdbCmsPortalList::GetList();
+
+        $restrictedNodes = [];
+        while ($portal = $portalList->Next()) {
+            $restrictedNodes[] = $portal->fieldMainNodeTree;
+            $navigationList = $portal->GetFieldPropertyNavigationsList();
+            while ($navigation = $navigationList->Next()) {
+                $restrictedNodes[] = $navigation->fieldTreeNode;
+            }
+        }
+
+        return $restrictedNodes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function GetHtmlHeadIncludes()
+    {
+        $includes = parent::GetHtmlHeadIncludes();
+        $includes[] = sprintf('<script src="%s" type="text/javascript"></script>', $this->getGlobal()->GetStaticURLToWebLib('/javascript/jsTree/3.3.8/jstree.js'));
+        $includes[] = sprintf('<script src="%s" type="text/javascript"></script>', $this->getGlobal()->GetStaticURLToWebLib('/javascript/navigationTree.js'));
+        $includes[] = sprintf('<script src="%s" type="text/javascript"></script>', $this->getGlobal()->GetStaticURLToWebLib('/javascript/jquery/cookie/jquery.cookie.js'));
+        $includes[] = sprintf('<link rel="stylesheet" href="%s">', $this->getGlobal()->GetStaticURLToWebLib('/javascript/jsTree/3.3.8/themes/default/style.css'));
+        $includes[] = sprintf('<link rel="stylesheet" href="%s">', $this->getGlobal()->GetStaticURLToWebLib('/javascript/jsTree/customStyles/style.css'));
+
+        return $includes;
+    }
+
+    /**
+     * @return TGlobal
+     */
+    private function getGlobal()
+    {
+        return ServiceLocator::get('chameleon_system_core.global');
```

Comment:
Please inject the service instead if loading it here.

Reply:
ok